src/libexpr/flake/flake.hh

```diff
@@ -102,7 +102,7 @@ struct LockFlags
 
     /* Whether to use the registries to lookup indirect flake
        references like 'nixpkgs'. */
-    bool useRegistries = true;
+    std::optional<bool> useRegistries = std::nullopt;
```

Comment:
A tristate is confusing to me, maybe it should just be initialized to `settings.registries`?

Reply:
I have thought about this a bit, I don't like initializing fields with global state. It may be confusing if the state changes after the structure has been initialized, since then `lockFlags.useRegistries` may go out of sync with `settings.useRegistries`.